sourcegraph/sourcegraph
# Add Code Insights edit dashboard UI

Closes https://github.com/sourcegraph/sourcegraph/issues/22231

### Background
This PR adds a new edit dashboard page on the `/insights/dashboard/:dashboardId/edit` route. Most of the UI is reused from the creation UI of the dashboard page. 

This PR is based on changes from https://github.com/sourcegraph/sourcegraph/pull/22564

Screenshot of new the dashboard edit page 
<img width="751" alt="Screenshot 2021-07-06 at 19 42 31" src="https://user-images.githubusercontent.com/18492575/124637408-64a41b00-de92-11eb-9578-73e1ae92c05f.png">

### How to test
1. Create the dashboard by dashboard creation UI (`/insights/add-dashboard` route)
2. Right after that you must see the edit dashboard UI by 
`/insights/dashboards/<camelCased name of your newly created dashboards>/edit`

### What have been done
- [x] Add new edit dashboard page (UI part) `/insights/dashboard/edit`
- [x] Add settings edit logic
